pkg/nodeagent/pod_controller.go

```diff
@@ -237,101 +244,152 @@ func buildRmdWorkload(pod *corev1.Pod) ([]*intelv1alpha1.RmdWorkload, error) {
-			case strings.HasSuffix(field, mbaPercentageConst):
-				mbaPercentage, err := strconv.Atoi(data)
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Mba.Percentage = mbaPercentage
-			case strings.HasSuffix(field, mbaMbpsConst):
-				mbaMbps, err := strconv.Atoi(data)
-				if err != nil {
-					continue
-				}
-				rmdWorkload.Spec.Rdt.Mba.Mbps = mbaMbps
-			case strings.HasSuffix(field, pstateRatioConst):
-				if data != "" {
-					rmdWorkload.Spec.Plugins.Pstate.Ratio = data
-				}
-			case strings.HasSuffix(field, pstateMonitoringConst):
-				if data != "" {
-					rmdWorkload.Spec.Plugins.Pstate.Monitoring = data
-				}
-			}
+		rmdWorkload.Spec.Policy = policy
+		rmdWorkload.Spec.Rdt.Cache.Min = minCache
+		rmdWorkload.Spec.Rdt.Mba.Percentage = mbaPercentage
+		rmdWorkload.Spec.Rdt.Mba.Mbps = mbaMbps
+		rmdWorkload.Spec.Plugins.Pstate.Ratio = pstateRatio
+		rmdWorkload.Spec.Plugins.Pstate.Monitoring = pstateMonitoring
 
-		}
 		rmdWorkloads = append(rmdWorkloads, rmdWorkload)
 	}
 	return rmdWorkloads, nil
 }
 
+func getAnnotationInfo(workloadData map[string]string, container corev1.Container) (policy string, minCache int, mbaPercentage int, mbaMbps int, pstateRatio string, pstateMonitoring string) {
+	var minCacheStr, mbaPercentStr, mbaMbpsStr string
+	for field, data := range workloadData {
+		if !strings.HasPrefix(field, container.Name) {
+			continue
+		}
+		switch {
+		case strings.HasSuffix(field, policyConst):
+			if data != "" {
+				policy = data
+			}
+		case strings.HasSuffix(field, cacheMinConst):
+			if data != "" {
+				minCacheStr = data
+			}
+		case strings.HasSuffix(field, mbaPercentageConst):
+			if data != "" {
+				mbaPercentStr = data
+			}
+		case strings.HasSuffix(field, mbaMbpsConst):
+			if data != "" {
+				mbaMbpsStr = data
+			}
+		case strings.HasSuffix(field, pstateRatioConst):
+			if data != "" {
+				pstateRatio = data
+			}
+		case strings.HasSuffix(field, pstateMonitoringConst):
+			if data != "" {
+				pstateMonitoring = data
+			}
+		}
+
+	}
+	minCache, err := strconv.Atoi(minCacheStr)
+	checkError(err)
+
+	mbaPercentage, err = strconv.Atoi(mbaPercentStr)
+	checkError(err)
+
+	mbaMbps, err = strconv.Atoi(mbaMbpsStr)
+	checkError(err)
+
+	return policy, minCache, mbaPercentage, mbaMbps, pstateRatio, pstateMonitoring
+}
+
+func checkError(err error) (result string) { //might remove this function in future
+	if err != nil { //only used 3 times
+		return err.Error()
+	}
+	return ""
+}
+
+func getContainerInfo(pod *corev1.Pod, container corev1.Container) containerInformation {
+	var containerInfo containerInformation //empty containerInformation struct
+
+	logger := log.WithName("buildRmdWorkload")
+	if !exclusiveCPUs(pod, &container) {
+		logger.Info("No container requesting cache found in pod")
+		containerInfo.coreIDs = nil
+		containerInfo.maxCache = 0
+		containerInfo.rmdWlStatus = nil
+		containerInfo.errStatus = nil
+		return containerInfo
```

Comment:
As per new semantics of the function, we can now simply:

````
return containerInfo{}, nil
````

Reply:
Should I be returning the struct containerInformation or containerInfo which is of type containerInformation? I really should've given more distinct names in hindsight...

Reply:
An empty `containerInformation{}` should suffice. So this should have actually been:
````
return containerInformation{}, nil
````